File: apps/portal-company/src/types/branch.ts
// src/types/branch.ts

export interface BranchBase {
  name: string;
  slug: string;
}

/** Payload para criação de filial */
export type BranchCreate = BranchBase;

/** Payload para atualização de filial */
export type BranchUpdate = BranchBase;

/**
 * Representação completa de uma filial retornada pela API
 */
export interface BranchRead extends BranchBase {
  id: string;          // UUID
  company_id: string;  // UUID da empresa
  created_at: string;  // ISO datetime
}

File: apps/backend/app/schemas/branch.py
### backend/app/schemas/branch.py ###
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel, ConfigDict

class BranchBase(BaseModel):
    name: str
    slug: str

class BranchCreate(BranchBase):
    pass

class BranchRead(BranchBase):
    id: UUID
    company_id: UUID
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)